R/model.R

```diff
@@ -506,6 +506,114 @@ compile_method <- function(quiet = TRUE,
 }
 CmdStanModel$set("public", name = "compile", value = compile_method)
 
+#' Check syntax of a Stan program
+#'
+#' @name model-method-syntax-check
+#' @aliases syntax_check
+#' @family CmdStanModel methods
+#'
+#' @description The `$syntax_check()` method of a [`CmdStanModel`] object
+#'   checks the Stan syntax of the model and returns `TRUE` if the compilation
+#'   succeeds. In case of a syntax error, an error is thrown.
+#'   
+#' @section Usage:
+#'   ```
+#'   $syntax_check(
+#'     quiet = TRUE,
```

Comment:
Unlike some of the other methods with a `quiet` argument, I'm not sure we need it here. Is there a case when it's helpful to see the command here? If not, I'd suggest removing the argument. If there is, then it's fine to keep it.

Reply:
quiet = FALSE now outputs "Stan program is syntactically correct"
quiet = TRUE only outputs anything on stderr from the compiler